runtime/memory.c

```diff
@@ -234,7 +234,9 @@ CAMLexport int caml_atomic_cas_field (
   } else {
     /* need a real CAS */
     atomic_value* p = &Op_atomic_val(obj)[field];
-    if (atomic_compare_exchange_strong(p, &oldval, newval)) {
+    int cas_ret = atomic_compare_exchange_strong(p, &oldval, newval);
+    atomic_thread_fence(memory_order_release); /* generates `dmb ish` */
```

Comment:
Again limiting my review to the memory model part, here is a minor nit. These C functions are not specific to Arm, and a lot more is needed to explain the hybrid scheme they implement now, so these comments are out of context. It _is_ important to explain why these fences are there, but perhaps it is better to have it as part of the upcoming memory model documentation?

Reply:
Would you be satisfied if the comment says 

> generates `dmb ish` on Arm64

Of course, the comment by itself is not illustrative unless accompanied by a memory model documentation, but will be correct. Once the documentation is available, we can update the comment to point to that.

Reply:
Whatever works for you, it was just an advice. It's fine to clarify it later.